Source/DedicatedServer/Private/ServerConsoleLinux.cpp

```diff
@@ -1,23 +1,244 @@
+
+						RedrawInputLine();
+					}
+				});
+			}
+			else if ( hInputEvents == '\b' )
+			{
+				if ( m_sInput.Len() >= 1 )
+				{
+					m_sInput.RemoveAt( m_sInput.Len() - 1 );
+					m_sInput.AppendChar(' ');
+					RedrawInputLine();
+					m_sInput.RemoveAt( m_sInput.Len() - 1 );
+					SetCursorPosition( coords( LINES - 1, m_sInput.Len() ) );
+				}
+			}
+			else if ( hInputEvents == 27 ) // Escape Key Event
+			{
+				ClearInputLine();
+				m_sInput.Empty();
+			}
+			else if ( hInputEvents == '\t' )
+			{
+				// ToDo: AutoCompletion
+			}
+			else if ( hInputEvents == KEY_UP )
+			{
+				if ( m_hCommandHistory.Num() == 0 ) return;
+
+				if ( m_iCommandHistoryIndex == -1 ) m_iCommandHistoryIndex = m_hCommandHistory.Num() - 1;
+				else --m_iCommandHistoryIndex;
+
+				if ( m_iCommandHistoryIndex < 0 ) m_iCommandHistoryIndex = 0;
+
+				if ( ( m_iCommandHistoryIndex + 1) == m_hCommandHistory.Num() && m_sUserInput.Len() == 0 ) m_sUserInput = m_sInput;
+
+				m_sInput = m_hCommandHistory[m_iCommandHistoryIndex];
+
+				RedrawInputLine();
+			}
+			else if ( hInputEvents == KEY_DOWN )
+			{
+				if ( m_iCommandHistoryIndex == -1 ) return;
+
+				++m_iCommandHistoryIndex;
+
+				if ( m_iCommandHistoryIndex > m_hCommandHistory.Num() ) m_iCommandHistoryIndex = m_hCommandHistory.Num();
+
+				if ( m_iCommandHistoryIndex == m_hCommandHistory.Num() )
+				{
+					if ( m_sUserInput.Len() > 0 ) m_sInput = m_sUserInput;
+					else m_sInput.Empty();
+
+					RedrawInputLine();
+				}
+				else
+				{
+					m_sInput = m_hCommandHistory[m_iCommandHistoryIndex];
+
+					RedrawInputLine();
+				}
+			}
+			else if ( hInputEvents == KEY_LEFT )
+			{
+				coords hCursorPosition(GetCursorPosition());
+
+				if ( hCursorPosition.X > 0 )
+				{
+					--hCursorPosition.X;
+					SetCursorPosition( hCursorPosition );
+				}
+			}
+			else if ( hInputEvents == KEY_RIGHT )
+			{
+				coords hCursorPosition( GetCursorPosition() );
+
+				if ( hCursorPosition.X < m_sInput.Len() )
+				{
+					++hCursorPosition.X;
+					SetCursorPosition( hCursorPosition );
+				}
+			}
+			else if ( hInputEvents == KEY_HOME )
+			{
+				coords hCursorPosition( GetCursorPosition() );
+				hCursorPosition.X = 0;
+				SetCursorPosition( hCursorPosition );
+			}
+			else if ( hInputEvents == KEY_END )
+			{
+				coords hCursorPosition( GetCursorPosition() );
+				hCursorPosition.X = m_sInput.Len();
+				SetCursorPosition( hCursorPosition );
+			}
+			else if ( hInputEvents == KEY_DC )
+			{
+				coords hCursorPosition( GetCursorPosition() );
+
+				if ( hCursorPosition.X <= m_sInput.Len() )
+				{
+					m_sInput.RemoveAt( hCursorPosition.X );
+					m_sInput.AppendChar( ' ' );
+
+					RedrawInputLine();
+
+					m_sInput.RemoveAt( m_sInput.Len() - 1 );
+
+					SetCursorPosition( hCursorPosition );
+				}
+			}
+			else if (hInputEvents == KEY_RESIZE)
+			{
+				refresh();
+			}
+			else
+			{
+				m_sInput.AppendChar( hInputEvents );
+				RedrawInputLine();
+			}
+		}
```

Comment:
Well mac has a lot more in common with linux than with windows, if you can make it so that those cooperate then mac will likely work too.

And unless someone who uses mac comes here and complains about issues I'd say just go for it (I don't have a mac to test stuff on too so it's probably going to take quite some time for this to become an actual issue)